Tests.Unit.Core.ApplicationServices/ApplicationServices/GDPR/DataProcessingRegistrationApplicationServiceTest.cs

```diff
@@ -1229,6 +1229,108 @@ public void Cannot_ClearBasisForTransfer_If_Write_Access_Is_Denied()
         }
 
+        [Fact]
+        public void Can_Update_IsOversightCompleted()
+        {
+            //Arrange
+            var id = A<int>();
+            var isOversightCompleted = A<YesNoUndecidedOption>();
+            var registration = new DataProcessingRegistration();
+            ExpectRepositoryGetToReturn(id, registration);
+            ExpectAllowModifyReturns(registration, true);
+            var transaction = new Mock<IDatabaseTransaction>();
+            _transactionManagerMock.Setup(x => x.Begin(IsolationLevel.ReadCommitted)).Returns(transaction.Object);
+
+            //Act
+            var result = _sut.UpdateIsOversightCompleted(id, isOversightCompleted);
+
+            //Assert
+            Assert.True(result.Ok);
+            Assert.Equal(isOversightCompleted, result.Value.IsOversightCompleted);
+            transaction.Verify(x => x.Commit());
+            _repositoryMock.Verify(x => x.Update(registration), Times.Once);
+        }
+
+        [Fact]
+        public void Update_IsOversightCompleted_Returns_Forbidden()
+        {
+            Test_Command_Which_Fails_With_Dpr_Insufficient_WriteAccess(id => _sut.UpdateIsOversightCompleted(id, A<YesNoUndecidedOption>()));
+        }
+
+        [Fact]
+        public void Update_IsOversightCompleted_Returns_Not_Found()
+        {
+            Test_Command_Which_Fails_With_Dpr_NotFound(id => _sut.UpdateIsOversightCompleted(id, A<YesNoUndecidedOption>()));
+        }
+
+        [Fact]
+        public void Can_Update_LatestOversightDate()
+        {
+            //Arrange
+            var id = A<int>();
+            var latestOversightDate = A<DateTime>();
+            var registration = new DataProcessingRegistration();
+            ExpectRepositoryGetToReturn(id, registration);
+            ExpectAllowModifyReturns(registration, true);
+            var transaction = new Mock<IDatabaseTransaction>();
+            _transactionManagerMock.Setup(x => x.Begin(IsolationLevel.ReadCommitted)).Returns(transaction.Object);
+
+            //Act
+            var result = _sut.UpdateLatestOversightDate(id, latestOversightDate);
+
+            //Assert
+            Assert.True(result.Ok);
+            Assert.Equal(latestOversightDate, result.Value.LatestOversightDate);
+            transaction.Verify(x => x.Commit());
+            _repositoryMock.Verify(x => x.Update(registration), Times.Once);
+        }
+
+        [Fact]
+        public void Update_LatestOversightDate_Returns_Forbidden()
+        {
+            Test_Command_Which_Fails_With_Dpr_Insufficient_WriteAccess(id => _sut.UpdateLatestOversightDate(id, A<DateTime>()));
+        }
+
+        [Fact]
+        public void Update_LatestOversightDate_Returns_Not_Found()
+        {
+            Test_Command_Which_Fails_With_Dpr_NotFound(id => _sut.UpdateLatestOversightDate(id, A<DateTime>()));
+        }
+
+        [Fact]
+        public void Can_Update_IsOversightCompletedRemark()
+        {
+            //Arrange
```

Comment:
Prøv og se hvordan de andre Success cases er lavet. Der er en hjælpe metode for at mindske kode duplikation